test/manual/api/improved_api_tests.sh

```
@@ -0,0 +1,676 @@
+
+  echo "[SCRIPT] Batch upsert systems: $batch_systems_json"
+  echo "[SCRIPT] Batch upsert connections: $batch_connections_json"
+
+  # Check for API_TOKEN
+  if [ -z "$API_TOKEN" ]; then
+    echo "❌ API_TOKEN is not set. Please export API_TOKEN before running the script."
+    return 1
+  fi
+
+  # Send batch upsert request
+  local response=$(curl -s -X POST "$API_BASE_URL/api/maps/$MAP_SLUG/systems" \
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)